charts/kube-state-metrics/templates/deployment.yaml

```diff
@@ -96,12 +96,17 @@ spec:
         {{- if .Values.kubeconfig.enabled }}
         - --kubeconfig=/opt/k8s/.kube/config
         {{- end }}
+        {{- if .Values.kubeRBACProxy.enabled }}
+        - --telemetry-host=127.0.0.1
+        - --telemetry-port={{ $telemetryPort }}
+        {{- else }}
```

Comment:
If rbac-proxy is enabled we only want to expose the telemetry endpoint through it